### DISCUSSION TITLE
keymon not catching keys in some windows.

### DISCUSSION THREAD

**Discussion by vdegenne** [WayfireWM/wayfire#2535] [Q&A] [answered]
I'm trying to screencast my keys using key-mon but in some terminal it won't work, here's a demo:

https://github.com/user-attachments/assets/e544bb12-ead1-454b-b0ee-3a6077bda3b6

What could be the reason behind that?
I remember having issues using a different layout in Alacritty too in the past.

**Comment by ammen99** [comment, marked as answer]
A guess: this keymon uses X11 keyboard grabs and they don't work in wayland-native apps :)

**Reply by vdegenne** [reply]
@ammen99 hey yes that was it. If I start Alacritty with `WAYLAND_DISPLAY` empty it works
